ops/pebble.py

```diff
@@ -538,6 +560,72 @@ def __repr__(self):
                 ).format(self=self)
 
 
+class FileType(enum.Enum):
+    """Enum of file types."""
+
+    FILE = 'file'
+    DIRECTORY = 'directory'
+    SYMLINK = 'symlink'
+    SOCKET = 'socket'
+    NAMED_PIPE = 'named-pipe'
+    DEVICE = 'device'
+    UNKNOWN = 'unknown'
+
+
+class FileInfo:
+    """Stat-like information about a single file."""
+
+    def __init__(
+        self,
+        path: str,
+        name: str,
+        type: Union['FileType', str],
+        size: Optional[int],
+        permissions: int,
+        last_modified: datetime.datetime,
+        user_id: Optional[int],
+        group_id: Optional[int],
+    ):
+        self.path = path
+        self.name = name
+        self.type = type
+        self.size = size
+        self.permissions = permissions
+        self.last_modified = last_modified
+        self.user_id = user_id
+        self.group_id = group_id
+
+    @classmethod
+    def from_dict(cls, d: Dict) -> 'FileInfo':
+        """Create new object from dict parsed from JSON."""
+        try:
+            file_type = FileType(d['type'])
+        except ValueError:
+            file_type = d['type']
+        return cls(
+            path=d['path'],
+            name=d['name'],
+            type=file_type,
+            size=d.get('size'),
+            permissions=int(d['permissions'], 8),
+            last_modified=_parse_timestamp(d['last-modified']),
```

Comment:
Do we need to be defensive in case 'last-modified' is malformed?
Perhaps _parse_timestamp() handles it already

Reply:
I don't think so -- in other places we just let the ValueError exception bubble up, which I think is the right thing to do. If Pebble is sending malformed data, something is seriously wrong (either a bug in our parsing code, or Pebble is really messed up).